### DISCUSSION TITLE
Using PR you can merge 1 change multiple times

### DISCUSSION THREAD

**Discussion by scrhicks** [community/community#66232] [Pull Requests]
### Select Topic Area

Bug

### Body

I'm trying do PR from one branch to another. Everything works fine but... you can repeat this step even you just merge the changes to the destination branch,

Step to reproduce:

1. Create repository with branch 'main'. (Add README.md file)
2. Create 'release' branch from 'main'
3. Add 'test1' file to 'main' branch
4. Create PR main into 'release' from 'main'
5. Merge PR using 'Squash and merge' button
6. Repeat step 4 and 5.

In this scenario you can crate infinite loop of the same commit into the specific branch.

Can someone tell me what is happen or what I'm doing wrong :D ?

P.S. I tested it using GitHub UI not the cli,

Best regards
scrhicks

**Comment by LeandroLuna** [comment]
Hi, Scrhicks!

In this scenario, when you create a new PR from 'main' into 'release' after merging the previous one with "Squash and merge," GitHub sees the commits on 'main' as new and not present in 'release,' so it will create a new commit (the squashed commit) and attempt to merge it again.

This behavior is not necessarily something you're doing wrong; it's more about how GitHub's "Squash and merge" option works. If you want to avoid this loop, you can consider different approaches:

- Rebase and merge: Instead of squashing and merging, you can use the "Rebase and merge" option when merging the PR. This option will rebase the changes from 'main' onto 'release,' preserving individual commits.

- Merge directly: If you don't need to squash the commits, you can simply use the "Create a merge commit" option when merging. This will create a merge commit that includes all the commits from 'main' into 'release' without squashing.

- Use a branch policy: You can implement branch protection policies in your repository settings to prevent PRs from being created that would introduce the same changes again, effectively preventing this loop.
